### PR TITLE
Improve centralization button behavior

### PR DESCRIPTION
**Pull request by je-l**
Fix bugs and improve overall styling.

Commits:

commit 674ff12a7efcccccc409b5988e4413ebbfc61efc

    Pan map position when clicking location button
    
    * Fix bug where QR code popup couldn't be closed
    
    * Fix bug where QR code popup could be opened in non-admin mode

commit ba8870a40b1d45202e0ac4c42aa301118497d682

    Use rounded buttons
    
    Also fix bug where public location modal was not displayed after user
    accepted location tracking.

commit 691cafee288a4b82cd40f845a9863f4487165b88

    More sensible handling for NaN values from server

### CODE REVIEW

File: client/src/3dVisualisation/screenContainer.tsx [hunk 1]
```diff
@@ -55,7 +55,9 @@ export const GenuineBusContainer = ({
           (topic: string, rawMessage: string) => {
             const parsed = parser.deserializeMessage(rawMessage);
 
-            screen.updateBeacons(parsed.map(i => mqttMessageToBabylon(i)));
+            if (parsed) {
```
Comment (reykjaviks) [inline client/src/3dVisualisation/screenContainer.tsx]: What is being parsed? Rename to improve readability.

File: client/src/admin/adminPanel.tsx [hunk 2]
```diff
@@ -107,6 +104,27 @@ interface Props {
   setNewHeight(a: string): void;
 }
 
+const HeaderRow = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const HideButton = styled.div`
```
Comment (reykjaviks) [inline client/src/admin/adminPanel.tsx]: Rename to CloseButton

File: client/src/map/marker.tsx [hunk 3]
```diff
@@ -68,14 +68,25 @@ export type PinKind = 'configure' | 'show' | 'none';
 interface PinProps {
   type: PinKind;
   coords: Location;
+  onClose(): void;
   onClick(a: MouseEvent<HTMLButtonElement>): void;
 }
 
-export const LocationPinMarker = ({ type, coords, onClick }: PinProps) => {
+export const LocationPinMarker = ({
```
Comment (reykjaviks) [inline client/src/map/marker.tsx]: Rename to LocationMarker or LocationPin.

File: client/src/router.tsx [hunk 4]
```diff
@@ -37,6 +37,27 @@ const MainRow = styled.div`
   height: 100%;
 `;
 
+export const isTrackingPromptOpen = (
+  bluetoothName: string | null,
+  shareLocationModalOpen: boolean,
+  publicShareOpen: boolean,
+  centralizeActive: boolean
```
Comment (reykjaviks) [inline client/src/router.tsx]: Add prefix "is" to all boolean variables and rename centralizeActive to isCentralizationButtonActive(?).

File: client/src/routerTest.tsx [hunk 5]
```diff
@@ -20,4 +21,36 @@ describe('<Router />', () => {
 
     mount(<Router />);
   });
+
+  it('shows tracking prompt if sharing location without known name', () => {
```
Comment (reykjaviks) [inline client/src/routerTest.tsx]: I am not quite sure what this sentence means. Please be more specific.

File: client/src/routerTest.tsx [hunk 6]
```diff
@@ -20,4 +21,36 @@ describe('<Router />', () => {
...
+        publicShareOpen,
+        centralizeActive
+      )
+    ).toBe(true);
+  });
+
+  it('shows tracking prompt if publishing location without known name', () => {
+    const bluetoothName = null;
+    const shareLocationModalOpen = false;
+    const publicShareOpen = true;
```
Comment (reykjaviks) [inline client/src/routerTest.tsx]: rename publicShareOpen to isUserSharingLocationPublicly or something similar.

File: client/src/common/navBar.tsx [hunk 7]
```diff
@@ -125,16 +122,12 @@ const NavBar = ({
   openShareLocationDropdown,
   openShareLocationModal,
 }: Props & RouteComponentProps) => {
-  const withBluetoothName = (after: () => void) => () => {
+  const withIndexPage = (after: () => void) => () => {
```
[the remark this answers is not in the document]
Comment (je-l) [reply client/src/common/navBar.tsx]: Renamed to `navigateHomeAndRun` as it describes the callback behavior.

**Pull request merged**